pkg/bootloader/grub.go

```diff
@@ -248,19 +249,21 @@ func (g *Grub) updateBootEntries(rootPath string, esp *deployment.Partition, new
 	activeEntries := []string{}
 
 	// Read current entries
-	stdOut, err := g.s.Runner().Run("grub2-editenv", grubEnvPath, "list")
-	if err != nil {
-		return fmt.Errorf("reading current boot entries: %w", err)
-	}
+	if ok, _ := vfs.Exists(g.s.FS(), grubEnvPath); ok {
+		stdOut, err := g.s.Runner().Run("grub2-editenv", grubEnvPath, "list")
+		if err != nil {
+			return fmt.Errorf("reading current boot entries: %w", err)
+		}
 
-	g.s.Logger().Debug("grub2-editenv stdout: %s", string(stdOut))
-	for line := range strings.SplitSeq(string(stdOut), "\n") {
-		if after, found := strings.CutPrefix(line, "entries="); found {
-			activeEntries = append(activeEntries, strings.Fields(after)...)
+		g.s.Logger().Debug("grub2-editenv stdout: %s", string(stdOut))
+		for line := range strings.SplitSeq(string(stdOut), "\n") {
+			if after, found := strings.CutPrefix(line, "entries=active "); found {
```

Comment:
Couldn't be a problem later to have `active` hardcoded here? As there is a constant for this maybe better to use `strings.CutPrefix(line, "entries="+DefaultBootID+" ")`?